### ISSUE TITLE
[Serializer > JSON Encoding] Allow configuring default options

### ISSUE THREAD

**Issue by dbu** [symfony/symfony#47183]
### Description

I need to configure json encoding options (in my case JSON_PRESERVE_ZERO_FRACTION). but the configuration for the `serializer` section does not allow to set options.

It would be nice to allow to configure options for the encoder and decoder in the configuration.

### Example

it would be nice if i could do something like this:
```
serializer:
    encoder: 
        options: JSON_PRESERVE_ZERO_FRACTION
    decoder: 
        options: JSON_THROW_ON_ERROR
```

my workaround is this compiler pass:
```
/**
 * Symfony serializer does not offer to configure options.
 */
class FloatSerializerPass implements CompilerPassInterface
{
    public function process(ContainerBuilder $container): void
    {
        $serializer = $container->getDefinition('serializer.encoder.json');
        $encode = new Definition(JsonEncode::class);
        $encode->setArgument(0, [JsonEncode::OPTIONS => \JSON_PRESERVE_ZERO_FRACTION]);
        $container->setDefinition('zero-encoder', $encode);
        $serializer->setArgument('$encodingImpl', new Reference('zero-encoder'));
    }
}
```

**Comment by schodemeiss** [comment]
The serializer *should* allow you to set the context for the JSON Encoder via the Conext Builder.

https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Serializer/Context/Encoder/JsonEncoderContextBuilder.php

Which should also means its customizable via the `framework -> serializer -> default_context` configuration, but alas, there is a bug which means that currently isn't working.

See for details: https://github.com/symfony/symfony/issues/47012

**Comment by nicolas-grekas** [comment]
We should also set `JSON_PRESERVE_ZERO_FRACTION` by default!

**Comment by schodemeiss** [comment]
> We should also set JSON_PRESERVE_ZERO_FRACTION by default!

Indeed - there's certainly a little work to be done around the serializer for 6.2. Unfortunately, I don't have the bandwidth to take a look at any of it :-(.

**Comment by dbu** [comment]
@nicolas-grekas i created the PR to set JSON_PRESERVE_ZERO_FRACTION by default. if #47012 is correct that it should be possible to configure the default context in that way, then this issue is actually a documentation request because it would already be possible to do what i wanted.

https://symfony.com/doc/current/reference/configuration/framework.html#serializer does not mention the default_context option of the serializer configuration.

**Comment by mpiot** [comment]
For the JSON Encoder/Decoder, the context from the configuration is not used, as described in https://github.com/symfony/symfony/issues/47012#issuecomment-1192372065

**Comment by schodemeiss** [comment]
> https://symfony.com/doc/current/reference/configuration/framework.html#serializer does not mention the default_context option of the serializer configuration.

Is *is* mentioned here, however: https://symfony.com/doc/current/serializer.html#serializer-context

**Comment by dbu** [comment]
the functionality to set configuration in the context has been implemented in #45554

i am adding documentation in https://github.com/symfony/symfony-docs/pull/17305
